web_stock/controllers/main.py

```diff
@@ -0,0 +1,226 @@
+            tpl_vals['error_fields'].append(
+                'Cannot find Picking Id %d' % picking_id,
+            )
+        else:
+            try:
+                request.env['website.stock.picking.wizard'].action_process_form(
+                    picking_id, kwargs,
+                )
+                tpl_vals.update({'id': picking_id.id})
+            except:
+                _logger.exception('Exception while saving picking')
+                tpl_vals['error_fields'].append(
+                    'Error while saving %s.' %  picking_id
+                )
+
+        return tpl_vals
+
+    def action_additional(self, picking_id, additional_action):
+        """ Additional actions provided by stock wizards """
+        picking_id = request.env['stock.picking'].browse(picking_id)
+        if additional_action == 'immediate_transfer':
+            wizard_id = request.env['stock.immediate.transfer'].create({
+                'pick_id': picking_id.id,
+            })
+            additional_action = 'process'
+            if picking_id.state == 'draft':
+                self.action_confirm(picking_id.id)
+        elif additional_action in ['process', 'process_cancel_backorder']:
+            wizard_id = request.env['stock.backorder.confirmation'].create({
+                'pick_id': picking_id.id,
+            })
+        return getattr(wizard_id, additional_action)()
+
+    def action_confirm(self, picking_id):
+        picking_id = self.__get_picking(picking_id)
+        return picking_id.action_confirm()
+
+    def action_assign(self, picking_id):
+        picking_id = self.__get_picking(picking_id)
+        return picking_id.action_assign()
+
+    def action_force_assign(self, picking_id):
+        picking_id = self.__get_picking(picking_id)
+        return picking_id.force_assign()
+
+    def action_new_transfer(self, picking_id):
+        picking_id = self.__get_picking(picking_id)
+        return picking_id.do_new_transfer()
+
+    def action_print_picking(self, picking_id):
+        picking_id = self.__get_picking(picking_id)
+        return picking_id.do_print_picking()
+
+    def action_cancel(self, picking_id):
+        picking_id = self.__get_picking(picking_id)
+        return picking_id.action_cancel()
+
+    def action_unreserve(self, picking_id):
+        picking_id = self.__get_picking(picking_id)
+        return picking_id.do_unreserve()
+
+    def action_put_in_pack(self, picking_id):
+        picking_id = self.__get_picking(picking_id)
+        return picking_id.put_in_pack()
+
+    def __get_picking(self, picking_id):
+        if isinstance(picking_id, int):
+            return request.env['stock.picking'].browse(picking_id)
+        else:
+            return picking_id
+
+    def __get_tpl_defaults(self, **kwargs):
+        return {
+            'errors': [],
+            'error_fields': kwargs.get('error_fields', []),
+            'db_info': json.dumps(db_info()),
+            'barcode_nomenclature': request.env.ref(
```

Comment:
When I try to set lots, this says that barcode.nomenclature(1,) is not serializable. I'm trying to set id, and update input